### PR TITLE
[ETH-FLOW-V2] Add contract addreses/abis/types

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Add `EthFlowContract` contract code to app
- include hook
- include contract types via updated typechain
- include contract addresses

# QA
UI
none.

Rest
- can just make sure nothing crazy is happening

### CODE REVIEW

File: src/custom/constants/index.ts [hunk 1]
```diff
@@ -44,6 +44,13 @@ export const APP_TITLE = 'CoW Swap | The smartest way to trade cryptocurrencies'
 // Smart contract wallets are filtered out by default, no need to add them to this list
 export const UNSUPPORTED_WC_WALLETS = new Set(['DeFi Wallet', 'WallETH'])
 
+export const COWSWAP_ETHFLOW_CONTRACT_ADDRESS: Partial<Record<number, string>> = {
```
Comment (W3stside) [inline src/custom/constants/index.ts:47]: will be updated in time

File: package.json [hunk 2]
```diff
@@ -205,6 +206,7 @@
   "license": "GPL-3.0-or-later",
   "dependencies": {
     "@amplitude/analytics-browser": "^1.1.4",
+    "@cowprotocol/ethflowcontract": "git+https://github.com/cowprotocol/ethflowcontract.git#a06e19c",
```
Comment (josojo) [inline package.json:209]: Good for now. Once we have the official releases, we should depend on release tags

**Review by josojo — APPROVED** [review]

File: src/custom/hooks/useContract/index.ts [hunk 3]
```diff
@@ -21,6 +26,16 @@ export * from './useContractMod'
 
 // Custom (non-MOD) hooks
 
+export function useEthFlowContract(): CoWSwapEthFlow | null {
```
Comment (shoom3301) [inline src/custom/hooks/useContract/index.ts:29]: Since this is a new code (not override), let's move it to `cow-react/modules/swap/hooks`

**Review by shoom3301 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (alfetopito) [reply package.json:209]: What about creating release candidates in the mean time?
Will look much cleaner

**Review by alfetopito — APPROVED** [review]

File: src/custom/hooks/useContract/index.ts (hunk 3 above)
Comment (W3stside) [reply src/custom/hooks/useContract/index.ts:29]: will move at the end in 1 pr :)

**Pull request merged**